diff --git a/dist/index.cjs b/dist/index.cjs
index 58583c81c7d53430a0fa0ec4272835d0612a9e0e..af680a98143fcb32435650df652c5e954395210b 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -144,8 +144,8 @@ async function getLastMatchingTag(inputTag) {
   const tags = await getGitTags();
   let tag;
   if (!isPrerelease2 && isVersion)
-    tag = tags.find((tag2) => tag2 !== inputTag && tag2[0] === "v" && !tag2.includes("-"));
-  tag || (tag = tags.find((tag2) => tag2 !== inputTag));
+    tag = tags.find((tag2) => tag2 !== inputTag && semver__default.valid(semver__default.coerce(tag)) && semver__default.prerelease(tag) === null);
+  tag || (tag = tags.find((tag2) => tag2 !== inputTag && semver__default.valid(semver__default.coerce(tag))));
   return tag;
 }
 async function isRefGitTag(to) {
diff --git a/dist/index.mjs b/dist/index.mjs
index 2aa6747fdfa40e0980f7cfc9480c99536e7c636b..d7587dbcbdeefb608a82b637f36adfd8feff7b68 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -138,8 +138,8 @@ async function getLastMatchingTag(inputTag) {
   const tags = await getGitTags();
   let tag;
   if (!isPrerelease2 && isVersion)
-    tag = tags.find((tag2) => tag2 !== inputTag && tag2[0] === "v" && !tag2.includes("-"));
-  tag || (tag = tags.find((tag2) => tag2 !== inputTag));
+    tag = tags.find((tag2) => tag2 !== inputTag && semver.valid(semver.coerce(tag)) && semver.prerelease(tag) === null);
+  tag || (tag = tags.find((tag2) => tag2 !== inputTag && semver.valid(semver.coerce(tag))));
   return tag;
 }
 async function isRefGitTag(to) {
